kernel_gateway/gatewayapp.py

```diff
@@ -213,17 +258,47 @@ def init_webapp(self):
         '''
         # Redefine handlers off the base_url path
         handlers = []
-        for handler in (
-            default_kernel_handlers + 
-            default_kernelspec_handlers +
-            default_base_handlers
-        ):
-            # Create a new handler pattern rooted at the base_url
-            pattern = url_path_join(self.base_url, handler[0])
-            # Some handlers take args, so retain those in addition to the
-            # handler class ref
-            new_handler = tuple([pattern] + list(handler[1:]))
-            handlers.append(new_handler)
+        if self.api:
```

Comment:
I know it's early and you haven't refactored yet, but I think the logic below should ultimately wind up in a function or functions defined in the `notebooks/handlers` module to encapsulate the parsing logic + execution logic in one spot? Invoke the function from here, but define it there.

Compelling reason to do it: I think these "handlers" are going to become part of a kind of lite plugin system. As long as you write a Python package/module with such-and-such API, you can plug it into kernel gateway to define new ways of talking to kernels.